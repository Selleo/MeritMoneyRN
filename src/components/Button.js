import React, { Component } from 'react'
import { Platform, TouchableOpacity, Text, StyleSheet, View } from 'react-native'
import LinearGradient from 'react-native-linear-gradient'
import { secondaryLight, primary, white, blueDark } from 'src/styles/colors'

export default class Button extends Component {
  static defaultProps = {
    onPress: () => {},
  }

  render() {
    const { text, outline, onPress } = this.props

    return (
      <TouchableOpacity onPress={onPress} style={styles.container}>
        <LinearGradient
          colors={[secondaryLight, primary]}
          end={{ x: 1, y: 1 }}
          locations={[0, 1]}
          start={{ x: 0, y: 1 }}
          style={styles.linearGradient}
        >
          <View style={[styles.textContainer, outline && styles.outline]} />
          <Text adjustsFontSizeToFit={true} style={styles.text}>
            {text}
          </Text>
        </LinearGradient>
      </TouchableOpacity>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    width: 150,
    height: 50,
  },
  linearGradient: {
    borderRadius: 25,
    zIndex: 1,
  },
  textContainer: {
    alignSelf: 'center',
    backgroundColor: 'transparent',
    borderRadius: 25,
    height: 42,
    position: 'absolute',
    top: 2.5,
    width: 142,
    zIndex: 2,
    ...Platform.select({
      ios: {
        height: 42,
      },
      android: {
        height: 38,
      },
    }),
  },
  outline: {
    backgroundColor: blueDark,
  },
  text: {
    textAlign: 'center',
    color: white,
    fontFamily: 'Lato-Regular',
    fontSize: 20,
    paddingVertical: 10,
    zIndex: 3,
  },
})
