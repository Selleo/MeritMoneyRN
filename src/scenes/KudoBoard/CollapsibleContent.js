import React, { PureComponent } from 'react'
import { StyleSheet, View } from 'react-native'

import Button from 'src/components/Button'
import StarRating from 'src/scenes/KudoBoard/StarRating'
import CommentInput from 'src/scenes/KudoBoard/CommentInput'

export default class CollapsibleContent extends PureComponent {
  state = {
    quantity: 1,
  }

  _handleTextChange = () => comment => this.setState({ comment })
  _handleQuantityChange = quantity => () => this.setState({ quantity })

  render() {
    const { comment, quantity } = this.state

    return (
      <View style={styles.container}>
        <StarRating quantity={quantity} setQuantity={this._handleQuantityChange} />
        <CommentInput handleChange={this._handleTextChange} value={comment} />
        <Button onPress={() => this.props.closeCollapsible(true)} text="GIVE KUDO" />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    flex: 1,
    paddingHorizontal: 10,
  },
})
