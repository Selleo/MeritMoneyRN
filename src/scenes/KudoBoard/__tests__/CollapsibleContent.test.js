import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import CollapsibleContent from '../CollapsibleContent'
import Button from 'src/components/Button'

describe('<CollapsibleContent />', () => {
  const defaultProps = {
    closeCollapsible: jest.fn(),
  }

  const wrapper = renderer.create(<CollapsibleContent {...defaultProps} />)

  test('render', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('handleTextChange', () => {
    test('should update comment state', () => {
      const instance = wrapper.getInstance()
      instance._handleTextChange()('test comment')
      expect(instance.state.comment).toEqual('test comment')
    })
  })

  describe('_handleQuantityChange', () => {
    test('should update kudos quantity', () => {
      const instance = wrapper.getInstance()
      instance._handleQuantityChange(5)()
    })
  })

  describe('Button onPress', () => {
    test('should call closeCollapsible with `true`', () => {
      const ButtonComponent = wrapper.root.findByType(Button)
      ButtonComponent.props.onPress()

      expect(wrapper.getInstance().props.closeCollapsible).toBeCalledWith(true)
    })
  })
})
