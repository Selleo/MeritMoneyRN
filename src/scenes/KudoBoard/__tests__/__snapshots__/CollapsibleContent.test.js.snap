// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CollapsibleContent /> render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "paddingHorizontal": 10,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginVertical": 5,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fcda53",
              "fontSize": 28,
            },
            Object {
              "paddingHorizontal": 8,
            },
            Object {
              "fontFamily": "icomoon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fcda53",
              "fontSize": 28,
            },
            Object {
              "paddingHorizontal": 8,
            },
            Object {
              "fontFamily": "icomoon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fcda53",
              "fontSize": 28,
            },
            Object {
              "paddingHorizontal": 8,
            },
            Object {
              "fontFamily": "icomoon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fcda53",
              "fontSize": 28,
            },
            Object {
              "paddingHorizontal": 8,
            },
            Object {
              "fontFamily": "icomoon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fcda53",
              "fontSize": 28,
            },
            Object {
              "paddingHorizontal": 8,
            },
            Object {
              "fontFamily": "icomoon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <BVLinearGradient
    colors={
      Array [
        4282147443,
        4281317757,
      ]
    }
    endPoint={
      Object {
        "x": 1,
        "y": 1,
      }
    }
    locations={
      Array [
        0,
        1,
      ]
    }
    startPoint={
      Object {
        "x": 0,
        "y": 1,
      }
    }
    style={
      Object {
        "borderRadius": 10,
        "marginVertical": 15,
        "zIndex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#222f3f",
          "borderRadius": 10,
          "height": 75,
          "position": "absolute",
          "top": 2.5,
          "width": 275,
          "zIndex": 2,
        }
      }
    />
    <TextInput
      allowFontScaling={true}
      multiline={true}
      onChangeText={[Function]}
      placeholder="I'd like to add a kudo because..."
      placeholderTextColor="#7f9d90"
      style={
        Object {
          "borderRadius": 10,
          "color": "#7f9d90",
          "fontSize": 16,
          "height": 80,
          "paddingHorizontal": 15,
          "paddingVertical": 3,
          "width": 280,
          "zIndex": 3,
        }
      }
      underlineColorAndroid="transparent"
    />
  </BVLinearGradient>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": 50,
        "opacity": 1,
        "width": 150,
      }
    }
  >
    <BVLinearGradient
      colors={
        Array [
          4282147443,
          4281317757,
        ]
      }
      endPoint={
        Object {
          "x": 1,
          "y": 1,
        }
      }
      locations={
        Array [
          0,
          1,
        ]
      }
      startPoint={
        Object {
          "x": 0,
          "y": 1,
        }
      }
      style={
        Object {
          "borderRadius": 25,
          "zIndex": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignSelf": "center",
              "backgroundColor": "transparent",
              "borderRadius": 25,
              "height": 42,
              "position": "absolute",
              "top": 2.5,
              "width": 142,
              "zIndex": 2,
            },
            undefined,
          ]
        }
      />
      <Text
        accessible={true}
        adjustsFontSizeToFit={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "Lato-Regular",
            "fontSize": 20,
            "paddingVertical": 10,
            "textAlign": "center",
            "zIndex": 3,
          }
        }
      >
        GIVE KUDO
      </Text>
    </BVLinearGradient>
  </View>
</View>
`;
